test/project_types/extension/features/argo/base_test.rb

```diff
@@ -0,0 +1,231 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'project_types/extension/extension_test_helpers'
+require 'base64'
+
+module Extension
+  module Features
+    module Argo
+      class BaseTest < MiniTest::Test
+        include TestHelpers::FakeUI
+        include ExtensionTestHelpers::Stubs::ArgoScript
+
+        def setup
+          super
+          ShopifyCli::ProjectType.load_type(:extension)
+
+          @dummy_argo = ExtensionTestHelpers::DummyArgoSetup.new
+          @result = 'fake result'
+          @error_message = "fake error"
+          @no_error = ''
+          @encoded_script = Base64.strict_encode64('var fake={}')
+        end
+
+        def test_config_aborts_with_error_if_script_file_doesnt_exist
+          File.stubs(:exist?).returns(false)
+          ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
+          Argo::Base.any_instance.stubs(:run_yarn_install).returns(true)
+          Argo::Base.any_instance.stubs(:run_yarn_run_script).returns(true)
+          @dummy_argo.stubs(:extract_argo_renderer_version).returns('0.0.1')
+          error = assert_raises ShopifyCli::Abort do
+            @dummy_argo.config(@context)
+          end
+
+          assert error.message.include?(@context.message('features.argo.missing_file_error'))
+        end
+
+        def test_config_aborts_with_error_if_script_serialization_fails
+          ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
```

Comment:
Not required, but you might be able to pull out some of these repeated stubs out into a private method so make the tests easier to read